models/opnilog/opnilog_parser.py

```diff
@@ -132,13 +135,24 @@ def train(
         ## train if no model
         model.train()
         logging.info(f"#######Training Model within {self.nr_epochs} epochs...######")
+        training_start_time = time.time()
         for epoch in range(self.nr_epochs):
-            logging.info(f"Epoch: {epoch}")
             self.run_epoch(
                 train_dataloader,
                 model,
                 SimpleLossCompute(model.generator, criterion, model_opt),
+                epoch=epoch,
+                training_start_time=training_start_time,
             )
+        end_time = time.time()
+        model_training_stats = {
```

Comment:
can we extract this (line 148-155) as a function -- as its reused at line 391 -- so we can reuse the code + make sure  they have same format.